lib/smart_proxy_reports/ansible_processor.rb

```diff
@@ -86,12 +105,68 @@ def process_facts(result)
       result["result"]["ansible_facts"] = {}
     end
 
+    # foreman-ansible-modules 3.0 does not contain summary field, convert it here
+    # https://github.com/theforeman/foreman-ansible-modules/pull/1325/files
+    def build_summary
+      if @data["summary"]
+        native = @data["summary"]
+      elsif (status = @data["status"])
+        native = {
+          "changed" => status["applied"] || 0,
+          "failures" => status["failed"] || 0,
+          "ignored" => 0,
+          "ok" => 0,
+          "rescued" => 0,
+          "skipped" => status["skipped"] || 0,
+          "unreachable" => 0,
+        }
+      else
+        native = {}
+      end
+      {
+        "foreman" => {
+          "change" => @change, "nochange" => @nochange, "failure" => @failure,
+        },
+        "native" => native,
```

Comment:
If we're going to use the native statuses/summary. Don't we need this for Puppet as well?

Reply:
Good point, let me add a copy too.